From fd028d16c9cec5a8bb63b856a898b295fe8585fe Mon Sep 17 00:00:00 2001
From: Vendula Poncova <vponcova@redhat.com>
Date: Sat, 4 Apr 2020 13:54:18 +0200
Subject: [PATCH] Don't use the default error register

Create a new instance of the error mapper, assign it to the message buses,
generate a new decorator for DBus errors and use it for the error registration.
---
 anaconda.spec.in                              |  2 +-
 pyanaconda/core/dbus.py                       | 21 +++++++++++++++----
 pyanaconda/modules/common/errors/__init__.py  |  4 +++-
 .../modules/common/errors/configuration.py    |  2 +-
 .../modules/common/errors/installation.py     |  2 +-
 pyanaconda/modules/common/errors/module.py    |  2 +-
 pyanaconda/modules/common/errors/payload.py   |  2 +-
 pyanaconda/modules/common/errors/storage.py   |  2 +-
 pyanaconda/modules/common/errors/task.py      |  2 +-
 9 files changed, 27 insertions(+), 12 deletions(-)

diff --git a/anaconda.spec.in b/anaconda.spec.in
index 6c48eae16..2e2560d8f 100644
--- a/anaconda.spec.in
+++ b/anaconda.spec.in
@@ -40,7 +40,7 @@ Source0: %{name}-%{version}.tar.bz2
 %define rpmver 4.10.0
 %define simplelinever 1.1-1
 %define utillinuxver 2.15.1
-%define dasbusver 0.2
+%define dasbusver 0.4
 
 BuildRequires: audit-libs-devel
 BuildRequires: libtool
diff --git a/pyanaconda/core/dbus.py b/pyanaconda/core/dbus.py
index d8542d9c7..2b7fea978 100644
--- a/pyanaconda/core/dbus.py
+++ b/pyanaconda/core/dbus.py
@@ -20,12 +20,13 @@ import os
 
 from dasbus.connection import SystemMessageBus, SessionMessageBus, MessageBus
 from dasbus.constants import DBUS_STARTER_ADDRESS
+from dasbus.error import ErrorMapper, get_error_decorator
 from pyanaconda.anaconda_loggers import get_module_logger
 from pyanaconda.core.constants import DBUS_ANACONDA_SESSION_ADDRESS, ANACONDA_BUS_ADDR_FILE
 
 log = get_module_logger(__name__)
 
-__all__ = ["DBus", "SystemBus", "SessionBus"]
+__all__ = ["DBus", "SystemBus", "SessionBus", "error_mapper", "dbus_error"]
 
 
 class AnacondaMessageBus(MessageBus):
@@ -71,11 +72,23 @@ class DefaultMessageBus(AnacondaMessageBus):
         return super()._find_bus_address()
 
 
+# The mapper of DBus errors.
+error_mapper = ErrorMapper()
+
 # Default bus. Anaconda uses this connection.
-DBus = DefaultMessageBus()
+DBus = DefaultMessageBus(
+    error_mapper=error_mapper
+)
 
 # System bus.
-SystemBus = SystemMessageBus()
+SystemBus = SystemMessageBus(
+    error_mapper=error_mapper
+)
 
 # Session bus.
-SessionBus = SessionMessageBus()
+SessionBus = SessionMessageBus(
+    error_mapper=error_mapper
+)
+
+# The decorator for DBus errors.
+dbus_error = get_error_decorator(error_mapper)
diff --git a/pyanaconda/modules/common/errors/__init__.py b/pyanaconda/modules/common/errors/__init__.py
index b4d2a6754..5d139c566 100644
--- a/pyanaconda/modules/common/errors/__init__.py
+++ b/pyanaconda/modules/common/errors/__init__.py
@@ -16,8 +16,10 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
+from dasbus.error import DBusError
+
 from pykickstart.errors import KickstartError
-from dasbus.error import DBusError, dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import ANACONDA_NAMESPACE
 
 __all__ = ["DBusError", "AnacondaError", "InvalidValueError"]
diff --git a/pyanaconda/modules/common/errors/configuration.py b/pyanaconda/modules/common/errors/configuration.py
index ef613f341..65975e7ef 100644
--- a/pyanaconda/modules/common/errors/configuration.py
+++ b/pyanaconda/modules/common/errors/configuration.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import ANACONDA_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
diff --git a/pyanaconda/modules/common/errors/installation.py b/pyanaconda/modules/common/errors/installation.py
index ecb805715..c7a656182 100644
--- a/pyanaconda/modules/common/errors/installation.py
+++ b/pyanaconda/modules/common/errors/installation.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import ANACONDA_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
diff --git a/pyanaconda/modules/common/errors/module.py b/pyanaconda/modules/common/errors/module.py
index 824a3b850..0160fb021 100644
--- a/pyanaconda/modules/common/errors/module.py
+++ b/pyanaconda/modules/common/errors/module.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import ANACONDA_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
diff --git a/pyanaconda/modules/common/errors/payload.py b/pyanaconda/modules/common/errors/payload.py
index be771e613..581d34157 100644
--- a/pyanaconda/modules/common/errors/payload.py
+++ b/pyanaconda/modules/common/errors/payload.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import PAYLOADS_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
diff --git a/pyanaconda/modules/common/errors/storage.py b/pyanaconda/modules/common/errors/storage.py
index cdd6e04aa..f18b80845 100644
--- a/pyanaconda/modules/common/errors/storage.py
+++ b/pyanaconda/modules/common/errors/storage.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import STORAGE_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
diff --git a/pyanaconda/modules/common/errors/task.py b/pyanaconda/modules/common/errors/task.py
index 08c5d5bcb..4d4c49914 100644
--- a/pyanaconda/modules/common/errors/task.py
+++ b/pyanaconda/modules/common/errors/task.py
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-from dasbus.error import dbus_error
+from pyanaconda.core.dbus import dbus_error
 from pyanaconda.modules.common.constants.namespaces import ANACONDA_NAMESPACE
 from pyanaconda.modules.common.errors import AnacondaError
 
-- 
2.25.4

